cocos/core/event/callbacks-invoker.ts

```diff
@@ -341,7 +375,14 @@ export class CallbacksInvoker {
      * 移除所有回调。
      */
     public clear () {
-        this._callbackTable = createMap(true);
+        for (let key in this._callbackTable) {
+            let list = this._callbackTable[key];
+            if (list) {
+                list.clear();
+                callbackListPool.free(list);
+                delete this._callbackTable[key];
```

Comment:
Is it safe too adjust the dictionary during iterating?

Reply:
It is, because the for loop use a prefetched local variable.

```
const infos = list.callbackInfos;
for (let i = 0, len = infos.length; i < len; ++i) {
}
```